packages/xml-views-validation/test/validators/attributes/unknown-attribute-key-spec.ts

```diff
@@ -0,0 +1,810 @@
+          expect(issues).to.deep.equal([
+            {
+              kind: "UnknownAttributeKey",
+              message: "Unknown attribute key: unknownattribute",
+              offsetRange: computeExpectedRange(xmlSnippet),
+              severity: "error",
+            },
+          ]);
+        },
+      });
+    });
+
+    it("will detect an invalid attribute key when the attribute doesn't have a value", () => {
+      const xmlSnippet = `
+          <mvc:View
+            xmlns:mvc="sap.ui.core.mvc"
+            xmlns="sap.m"
+            🢂busy1🢀 >
+          </mvc:View>`;
+
+      testValidationsScenario({
+        model: ui5SemanticModel,
+        xmlText: xmlSnippet,
+        validators: {
+          attribute: [validateUnknownAttributeKey],
+        },
+        assertion: (issues) => {
+          expect(issues).to.deep.equal([
+            {
+              kind: "UnknownAttributeKey",
+              message: "Unknown attribute key: busy1",
+              offsetRange: computeExpectedRange(xmlSnippet),
+              severity: "error",
+            },
+          ]);
+        },
+      });
+    });
+
+    it("will detect an invalid attribute key in aggregation element", () => {
+      const xmlSnippet = `
+          <mvc:View
+            xmlns:mvc="sap.ui.core.mvc"
+            xmlns="sap.m">
+            <content 🢂unknownattribute🢀="">
+              <List></List>
+            </content>
+          </mvc:View>`;
+
+      testValidationsScenario({
+        model: ui5SemanticModel,
+        xmlText: xmlSnippet,
+        validators: {
+          attribute: [validateUnknownAttributeKey],
+        },
+        assertion: (issues) => {
+          expect(issues).to.deep.equal([
+            {
+              kind: "UnknownAttributeKey",
+              message: "Unknown attribute key: unknownattribute",
+              offsetRange: computeExpectedRange(xmlSnippet),
+              severity: "error",
+            },
+          ]);
+        },
+      });
+    });
+
+    it("will detect an invalid xmlns attribute key element", () => {
+      const xmlSnippet = `
+                <mvc:View
+                 xmlns:mvc="sap.ui.core.mvc"
+                 🢂xmlns:m:1🢀="sap.m">
+                </mvc:View>`;
+
+      testValidationsScenario({
+        model: ui5SemanticModel,
+        xmlText: xmlSnippet,
+        validators: {
+          attribute: [validateUnknownAttributeKey],
+        },
+        assertion: (issues) => {
+          expect(issues).to.deep.equal([
+            {
+              kind: "UnknownAttributeKey",
+              message: "Unknown attribute key: xmlns:m:1",
+              offsetRange: computeExpectedRange(xmlSnippet),
+              severity: "error",
+            },
+          ]);
+        },
+      });
+    });
+
+    it("will detect 'stashed' as invalid in aggregation element", () => {
+      const xmlSnippet = `
+      <mvc:View
+        xmlns:mvc="sap.ui.core.mvc"
+        xmlns="sap.m">
+        <content 🢂stashed🢀="">
```

Comment:
There is a limit of use cases we can reasonably check, maybe using maps/arrays for whitelists instead of conditionals will reduce the cyclomatic complexity and thus reduce the tests needed for 100% coverage.

Reply:
Even if I extract the special cases to a list I would still like to test specific cases to make sure we didn't miss anything

Reply:
its o.k. now, but if we have 100 special cases we cannot reasonably test all..
- Maybe with generated tests